client/src/components/Ticketing/ticketingmanager/DoorList/DoorList.tsx

```diff
@@ -6,72 +6,71 @@ import {titleCase} from '../../../../utils/arrays';
 import {useAuth0} from '@auth0/auth0-react';
 import {toDateStringFormat} from '../Event/components/util/EventsUtil';
 import format from 'date-fns/format';
+import {useFetchToken} from '../Event/components/ShowingUtils';
 
-/**
- * Used to check the guests in
- *
- * @param {boolean} isCheckedIn
- * @param {string} ticketID
- * @returns
- */
-const checkInGuest = async (isCheckedIn: boolean, ticketID: string) => {
-  const {getAccessTokenSilently} = useAuth0();
-  try {
-    const token = await getAccessTokenSilently({
-      audience: process.env.REACT_APP_ROOT_URL,
-      scope: 'admin',
-    });
-    const res = await fetch(
-      process.env.REACT_APP_API_2_URL + `/events/checkin`,
-      {
-        credentials: 'include',
-        method: 'PUT',
-        headers: {
-          'Content-Type': 'application/json',
-          'Authorization': `Bearer ${token}`,
-        },
-        body: JSON.stringify({isCheckedIn, ticketID}),
-      },
-    );
-    if (!res.ok) {
-      throw new Error(`Failed to check in guest. HTTP status: ${res.status}`);
-    }
-    return res.json();
-  } catch (err) {
-    console.error(err.message);
-  }
-};
 
 const renderCheckbox = (params: GridCellParams) => (
   <Checkbox checked={params.value as boolean} disabled color='info' />
 );
 
-/**
- * renders in the check in for guests
- *
- * @param {GridCellParams} params
- * @returns edits the checkInGuest value
- */
-const renderCheckin = (params: GridCellParams) => {
-  const isDisabled = params.getValue(params.id, 'lastname') === 'OPEN SEATS';
+interface RenderCheckinProps {
+  params: GridCellParams;
+}
+
+const RenderCheckin = (props: RenderCheckinProps) => {
+  const {params} = props;
+  const {token} = useFetchToken();
```

Comment:
Instead of fetching a token for each Checkin, would it make more sense to fetch 1 token in the doorlist component, and then pass it in through the props?

Reply:
That makes sense, I just have to move the columns definition into the scope of the doorlist component.  👍

Reply:
Fixed